### PR TITLE
fix(gates): gate cards show the evaluator verdict, floor results and denial remedy; delivery strip ignores non-delivering workflows (#250)

### LINKED ISSUE #250: gate cards never show the evaluator verdict/criterion (F-036 UI half); home delivery strip counts onboarding runs as "Vacuous — needs retry"
## Context

Found by the Phase 3 re-run of the acceptance program (fresh rig: wicked-crew 0.7.28, studio bundle 0.5.4, core-ts 0.7.18; governed `bug` run `d74e4e8f-bbc9-4697-8c30-181bae005217` on wicked-studio#214, delivered as wicked-studio#249 which stays open as evidence). Evidence dir on the rig: `/private/tmp/fresh/evidence/phase3-r2/` (file names below are relative to it). Format of each finding: id | phase | severity | surface | what | evidence | proposed fix.

### F-3R2-006

```
F-3R2-006 | phase3-r2 | MEDIUM | studio gate card (F-036 UI half) | Neither the verify pre-run gate card (the "fix gate" the operator answers) nor the deliver 
gate card shows the evaluator verdict: `verdict-detail` blocks = 0, no "Evaluator verdict" text, no criterion on the card. The verdict exists only as the 
thread line "Checks ran on fix — pass" (details on expand). At the DENIED verify gate the card said only "Unit 4 verdict is NOT PASS — confirm to retry the 
phase, or reject to cancel the run"; the reason (evaluator≠creator, the changed path, the restore command) was only in the thread. | g4-card.png, 
g4-gate-card-text.txt, g5-card.png, g5-gate-card-text.txt, g6-card.png, bundle `cM`/`verdict-detail` renders only inside a gateEvaluated thread entry | render 
the last gateEvaluated (criterion, floor pass/fail, agent verdict + reasoning, denial reason/source) on the gate card itself; for a denial show the remedy and 
what Approve will do
```

### F-3R2-018

```
F-3R2-018 | phase3-r2 | MEDIUM | home page delivery strip (misleading copy) | After the run the landing strip reads "2 Verified & delivered · 0 Stranded — 
needs review · 9 Vacuous — needs retry": the 9 "vacuous" runs are the onboarding runs, which deliver nothing by design; "needs retry" is wrong and buries the 
one real signal. | 52-home-after-run.png, console.log 23:20:37 | exclude non-delivering workflows (onboarding, doc/demo) from the vacuous bucket or label them 
"no delivery expected"
```

## Ask

Fix these together in one PR (they share a surface); the proposed fixes in each finding are the coordinator's suggestion, not a constraint. No version bump. Refs wave 3 of the fix program (2026-09-10).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by mikeparcewski**
Closes #250

Fixes the two Phase 3 re-run findings that share the gate/delivery surface (wave 3 of the fix program, 2026-09-10). No version bump.

## F-3R2-006 — gate cards show the evaluator verdict, the floor results and the denial remedy

**Before.** The verify pre-run gate card read only "Approve unit 4 before it runs: verify — …" while the `fix` phase's PASS (criterion, deterministic floor, judge reasoning) sat in the already-hydrated event log; the DENIED gate read "Unit 4 verdict is NOT PASS — confirm to retry the phase, or reject to cancel the run" while the reason (evaluator≠creator, the changed path, the restore command) was only in an expandable thread line; the deliver gate never showed that the repository's own checks had run.

**After.** `SteeringGate` renders a `gate-verdict` block between the question and the answer controls, from the run's own event log (`useRunEventStore` — the same source `VerdictDetail` reads; **zero new requests**):

- **Which evaluation:** the last `gateEvaluated` at or below the gate's `ord` — for a pre-run gate "before unit #N" that is the phase that just finished (the fix gate); for an escalated gate it is unit N's own denial. No evaluation yet ⇒ **no block**, never a verdict fabricated from the prompt.
- **Verdict + criterion + judge:** `PASS` / `DENIED` / `UNGATED` (an evaluation with no floor, no judge and no policy is labelled a default-allow, not a pass — FINDING-025), the `criterion`, `judge: <agentVerdict> — <agentReasoning>`, and the layers line (`deterministic floor: pass|fail · evaluator: …`).
- **Floor (F-039):** the `repoChecksEvaluated` frame from the same fold — one row per check: `typecheck · exit 0 · 6.9s · package.json scripts.typecheck`, timed-out / could-not-start spelled out, `skipped (an earlier check failed): test`, and the two `checks: []` cases disclosed ("no detectable check — a disclosed vacuous pass" vs "the floor failed before any check ran").
- **Denial:** `denied by <layer>: <engine reason verbatim>` from `GateEvaluatedEvent.denial` (`data-denial-source="worktree_guard"` etc.); backticked spans in the reason render as copyable `<code>`, so the remedy `git read-tree --reset -u <tree>` and the def change `executes_code: true` are right on the card. An older engine that sends only `denialReason` prose still gets a denial view (`source: null`).
- **Mutation record (F-036):** `worktree changed by pi during verify: M src/App.tsx (tree a1b2c3d4e5 → f0e1d2c3b4)`, plus "the run branch moved" when `headMoved`.
- **Evidence belongs to one fold:** the backward search for `repoChecksEvaluated` / `evaluatorMutatedWorktree` stops at the previous same-ord `gateEvaluated`, so a retry's PASS never inherits the replaced attempt's changed path (the recorded G6 case).
- **Every gate card, not just the run page's:** the project/landing gate inbox (`CenterDashboard`'s `GateActionCard`) renders the same `GateVerdict` block from the event log it already subscribes to (Copilot follow-up on this PR). `ApprovalDock` passes `view.units` so the phase is named (`fix`, `verify`); `phaseLabel` is now shared with `VerdictDetail` (moved, not duplicated). The e2e suites' selectors (`steering-gate`, `steering-prompt`, `verdict-detail`) are untouched — the new block has its own `gate-verdict-*` testids; inventory regenerated.
- **Always bounded (Copilot follow-ups):** the lookup requires a numeric gate ord — an ord-less `gateEvaluated` is never selected for a bounded gate, and a card with no ord renders no block rather than an unbounded "last evaluation". In the daemon-restart fallback (status `awaiting_human`, transient gate cache lost) `ApprovalDock` derives the ord from the run's own cursor exactly as `useRunModel.pendingGate` does (`units[unit_ix].ord`, else `unit_ix + 1`), so the run page's card stays bound even when the prompt was lost — and the cursor indexes units in ORD order, so the DTO's array is sorted before `unit_ix` is applied (the contract `ChatPanel`'s render already states). `formatDuration` rounds to the unit before choosing the format (no `60.0s` / `1m 60s`).
- **Persisted verdicts survive a reload (Copilot follow-up):** only the run page's route hydrates `/runs/:id/events`, so after a landing/project reload the inbox card would have had nothing to read. `CenterDashboard` now backfills the event log for OPEN-GATE runs only — once per GATE INSTANCE (`runId:ord:receivedAt`) per mount, so a run that opens a second gate while the dashboard stays mounted (a retry of the same ord included) refreshes its durable prefix and a reconnect gap between the two can never leave the earlier verdict standing under the new gate — the same bounded pattern `useBoardModel` uses for failed runs, so the list surface's request budget is O(open gates), not O(rows). No "already has frames" shortcut — after a reconnect the live slice may hold only the new `awaitingHuman` while the `gateEvaluated` was recorded before the socket came up, so the once-per-id guard alone bounds the fetch; the effect depends on `openGates` only (the store is read inside), so a busy dashboard does not rescan its gates on every structured frame. **Readiness per gate instance (Copilot follow-up):** while an instance's fetch is pending the inbox card renders NO verdict — the store's slice may be an earlier attempt's evaluation plus this gate's `awaitingHuman`, so on a same-ord retry after a reconnect "render what we have" would have shown the OLD verdict under the new gate; an instance becomes ready only when its fetch resolves with history, and an empty history or a 503 leaves it un-ready for good (no block, never a stale one). `hydrate` merges live frames by fingerprint so nothing double-counts.
- **Denial spellings (Copilot follow-up):** the structured denial is read in both casings — `claimId`/`claim_id`, `ruleIds`/`rule_ids`, `deniedTool`/`denied_tool` — the same tolerance `denialCopy.ts` already keeps, so a snake_case engine never drops its rule ids from the card.

**Wire contract:** `wicked-crew-api-types` **0.30.0 → 0.31.0** (devDependency, exact pin). Verified purely additive against the installed 0.30.0 `index.d.ts` (one doc-comment line changed, 151 lines added: `UnitDenialSource`, `UnitDenial`, `GateEvaluatedEvent.denial`/`evaluatorPolicies`, `WorktreeChangedPath`, `EvaluatorMutatedWorktreeEvent`, `RepoCheckRun`, `RepoChecksEvaluatedEvent`, `GateEvidenceEvent`); `tsc --noEmit` green on the bump alone before any source change. Every field the card renders is declared in 0.31.0; nothing is invented.

## F-3R2-018 — the delivery strip ignores non-delivering workflows

**Before.** "2 Verified & delivered · 0 Stranded — needs review · **9 Vacuous — needs retry**" — the nine were `onboarding` runs (verified against the rig's baseline: 9 × `onboarding`, 2 × `bug`), which the daemon stamps `delivery: 'vacuous'` because an untouched worktree is that workflow's designed outcome.

**After.** `deliveryCounts(runs, isSystemWorkflow?)` licenses the `vacuous` bucket with the Delivery section's existing `canDeliver` rule — a deliver unit on the run (evidence first: a materialised `wf-<id>` def whose deliver phase ran and found nothing still counts), or a workflow positively known not to be a system one (`is_system` from the ONE budgeted `GET /workflows`; `onboarding` is also on the denylist, so the count is right on a cold catalog too). Both consumers — `DeckVerifiedStrip` and the KPI ribbon's Review tile — pass the same lookup, so they cannot disagree. The cell reads **"Vacuous — no change to deliver"**: the condition, not a prescription. The licence can only withhold a count, never invent one (a bare `vacuous` stamp on an unclassifiable run is not shown — the same trade `canDeliver` already makes for the Delivery section).

## Tests

Recorded frames from the Phase 3 re-run (`GET /runs/:id/events` at G4 / G5 / G6 plus the `repoChecksEvaluated` capture) as a fixture, `tests/fixtures/gateEvidence.ts` — the gate-relevant subset of the recording with the recording's `seq`/`ts` and every rendered field byte-for-byte (the reviewer's W3S-252-01 is taken: the G6 ord-4 `gateEvaluated` now carries the recorded joined `criterion` and `agentReasoning`, and the connective frames between gates are the recorded `resumed` / `unitDispatched` / `unitExecuting` / `unitReworkAmended`, not invented ones). Scrubbed: run id, intent text (prompts and the amendment's re-stated description), worktree paths, test-runner tails, and the two git tree ids (synthetic hex of the same width, the denial prose and the amendment kept in their exact wording). One synthetic frame (`REPO_CHECKS_FAIL`, labelled) pins the failed-floor rendering in the wire's declared shape. W3S-252-02 is taken too: `overflow-wrap: anywhere` on `gate-verdict-denial`, so the 40-hex tree id inside one `<code>` span wraps in a narrow dock.

- `tests/gateVerdictModel.test.ts` (16) — which evaluation answers which gate, the ord bound (incl. ord-less frames), same-fold evidence, ungated classification, prose-only and snake_case denials, malformed-frame narrowing, helpers (incl. duration boundaries).
- `tests/SteeringGate.verdict.test.tsx` (8) — the card over G4 / G5 / G6 / failed floor / first gate / ungated / no units / no ord.
- `tests/CenterDashboard.build.test.tsx` (+8) — the gate inbox card renders the same verdict block (G4), none without an evaluation, after a reload it backfills the open-gate run's log exactly once (a gate-less run is never fetched), after a reconnect (only the live `awaitingHuman` in the store) the backfill still runs and the live frame is merged once, the guard is per gate instance (a re-set of the same gate fetches nothing, a second gate on the same run fetches exactly once more and proves its own history before its block renders), and the readiness rule is pinned three ways — a same-ord retry with a deferred fetch renders no block while pending and the retry's own `repo_checks` denial once the log lands (never attempt 0's worktree-guard denial), an empty history ⇒ no block, a 503 ⇒ no block; `tests/ChatPanel.dock.test.tsx` (+2, +1 assertion) — the daemon-restart fallback derives the ord (from an unsorted DTO too) and keeps the lookup bounded.
- `tests/deliveryCounts.test.ts` (6) — the recorded nine-onboarding case (warm, cold, no lookup), evidence-first deliver unit, positively-known build, withheld-when-unknown, other system workflows, delivered/stranded/archived untouched.
- `tests/DeckVerifiedStrip.test.tsx` (+2, existing fixture made a licensed vacuous run), `tests/DeckKpiRibbon.test.tsx` (fixture likewise).

`npm run lint` · `npm run typecheck` · `npm test` (256 files, 2781 tests) · `npm run build` — all green locally on the fresh clone. The `e2e/` suites are Python Playwright against a live daemon and were not run (no daemon may be touched from this wave); the selectors they key on are unchanged.

## Wire gaps (rendered what exists; listed rather than invented)

- **Judge seat** — `gateEvaluated` (api-types 0.31.0) carries no seat/CLI for the layer-2 judge; the brief asked for `judge: <seat>` when present. Not present ⇒ nothing rendered, and deliberately not inferred from `unitDispatched.cli` (that is the creator's seat, not the judge's). A `judgeCli?: string` on `GateEvaluatedEvent` would light this up with a one-line change.
- **`awaitingHuman.reviewingOrd`** — on the recorded wire (`null` for pre-run gates, `4` for the escalation) but not declared in api-types. Not needed: the card keys on the gate's `ord` and log order instead. Declaring it would let the card distinguish "before unit N" from "N's verdict escalated" without reading the prompt.
- **Fail-closed worktree-guard denials** carry no `evaluatorMutatedWorktree` (by design per the 0.31.0 docs); the card then shows the `denial.reason`, which names the cause. No gap in the card, noted for completeness.

## Declined / out of scope (with reason)

- The strip's `?filter=vacuous|stranded|delivered` doors land on `WorkPage`'s "All" tab — `routedFilter` honours only the five status tabs. Pre-existing and unrelated to either finding; left as is rather than growing this PR into a Work-page filter feature.
- `stranded` is not licensed the way `vacuous` now is: the daemon's `'stranded'` is its own positive worktree-carries-work check, and the finding is about the vacuous bucket only.
- No narrator line for `repoChecksEvaluated` / `evaluatorMutatedWorktree` — the finding is the gate card; the feed's existing "Checks ran on <phase> — pass|deny" line stands.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/gateVerdictModel.ts [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  return { word: c.exitCode === 0 ? 'exit 0' : `exit ${c.exitCode}`, ok: c.exitCode === 0 };
+}
+
+/** `6893` → `6.9s`, `79191` → `1m 19s`, `420` → `420ms`. */
+export function formatDuration(ms: number): string {
+  if (ms < 1000) return `${Math.max(0, Math.round(ms))}ms`;
+  if (ms < 60_000) return `${(ms / 1000).toFixed(1)}s`;
+  const m = Math.floor(ms / 60_000);
+  const s = Math.round((ms - m * 60_000) / 1000);
+  return s === 0 ? `${m}m` : `${m}m ${s}s`;
```
[the remark this answers is not in the document]
Comment (mikeparcewski) [reply src/components/gateVerdictModel.ts]: Agreed — the total was rounded after the unit was picked, so 59,999 ms read "60.0s" and 119,500 ms read "1m 60s". Fixed in c508f17: `formatDuration` now rounds to the unit it is about to print (tenths under a minute, whole seconds above) BEFORE choosing the format — 59,999 → `1m`, 119,500 → `2m`, 119,499 → `1m 59s`, 999.6 → `1.0s`. Boundary cases pinned in `tests/gateVerdictModel.test.ts`.

File: src/components/gateVerdictModel.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+ * The deciding evaluation for a gate on `gateOrd` (the card's `ord`; omit to take the last
+ * evaluation in the log), with its floor and mutation evidence attached. `null` when the log
+ * holds no qualifying `gateEvaluated`.
+ */
+export function gateVerdict(events: readonly CoreEvent[], gateOrd?: number): GateVerdictView | null {
+  let idx = -1;
+  for (let i = 0; i < events.length; i++) {
+    const e = events[i]!;
+    if (e.type !== 'gateEvaluated') continue;
+    if (gateOrd !== undefined && typeof e.ord === 'number' && e.ord > gateOrd) continue;
```
[the remark this answers is not in the document]
Comment (mikeparcewski) [reply src/components/gateVerdictModel.ts]: Agreed — a bounded lookup has nothing to bound against when the frame carries no numeric `ord`, and the old condition let such a frame through as "≤ gateOrd". Fixed in 22fb9db: with `gateOrd` supplied, `gateVerdict` now skips any `gateEvaluated` whose `ord` is not a number (the unbounded path is unchanged). Pinned in `tests/gateVerdictModel.test.ts`: an ord-less frame appended after G4 no longer displaces the ord-3 verdict, alone in the log it yields `null` for a bounded gate, and unbounded still takes it.

**Comment by mikeparcewski** [comment]
Re Copilot's third pass ("Needs a closer look" — two suppressed remarks, no threads to resolve), both taken in 946570c:

1. **Landing/project gate inbox** — `CenterDashboard`'s `GateActionCard` rendered only the prompt, so F-3R2-006 stayed open on that surface. It now renders the same `GateVerdict` block from the event log the dashboard already subscribes to (`byRun[gate.runId]`, zero new requests), bounded on the gate's ord, with the run's units naming the phase. Pinned in `tests/CenterDashboard.build.test.tsx` (G4 fixture ⇒ `gate-verdict[data-verdict=pass]` inside `gate-inbox`; no evaluation ⇒ no block).
2. **Daemon-restart fallback / unbounded lookup** — `ApprovalDock` now derives the ord from the run's own cursor exactly as `useRunModel.pendingGate` does (`units[unit_ix].ord`, else `unit_ix + 1`), and `SteeringGate` renders the block only with a numeric ord — never an unbounded "last evaluation". Pinned in `tests/ChatPanel.dock.test.tsx` (fallback card reads `before unit #2`; a stray later-ord frame is not shown) and `tests/SteeringGate.verdict.test.tsx` (no ord ⇒ no block even with a full log).

Lint / typecheck / full suite (256 files, 2773 tests) / build green locally; PR body updated accordingly.

**Comment by mikeparcewski** [comment]
Re Copilot's fourth pass (two suppressed remarks, no threads to resolve), both taken in fd87f27:

1. **`unit_ix` on an unsorted DTO** — the cursor indexes units in ORD order and `SessionView.units` is not guaranteed sorted, so `ApprovalDock`'s restart fallback now sorts by `ord` before applying `unit_ix` (the contract `ChatPanel`'s render already states). Pinned in `tests/ChatPanel.dock.test.tsx`: a build-first DTO with `unit_ix: 1` still binds the card to `before unit #2`.
2. **Inbox card after a reload** — only the run page's route hydrated `/runs/:id/events`, so a persisted evaluation was absent from the inbox card. `CenterDashboard` now backfills the event log for OPEN-GATE runs only — once per run id per mount, skipped when frames are present — the same bounded pattern `useBoardModel` uses for failed runs (budget O(open gates), not O(rows)); degrades silently on an api surface without `getRunEvents`, a 503, or an empty history, and `hydrate` merges live frames by fingerprint. Pinned in `tests/CenterDashboard.build.test.tsx`: empty store + mocked durable log ⇒ the G4 verdict renders; `getRunEvents` called exactly once, for the gated run only.

Lint / typecheck / full suite (256 files, 2775 tests) / build green locally; PR body updated.

File: src/components/CenterDashboard.tsx [hunk 3]
```diff
@@ -963,6 +984,32 @@ export function CenterDashboard({
...
+  // way `useBoardModel`'s failed-run backfill is: once per run id per mount, and only for runs
+  // that currently hold a gate (a paused run has exactly one), so the list surface's request
+  // budget stays O(open gates), not O(rows). Degrades silently — an api surface without
+  // `getRunEvents`, a 503 (no event-log binding) or an empty history leaves the card promptless,
+  // never wrong; `hydrate` merges live frames by fingerprint, so nothing is double-counted.
+  const gateBackfilled = useRef<Set<string>>(new Set());
+  useEffect(() => {
+    for (const g of openGates) {
+      const id = g.runId;
+      if (gateBackfilled.current.has(id) || (byRun[id] ?? []).length > 0) continue;
```
[the remark this answers is not in the document]
Comment (mikeparcewski) [reply src/components/CenterDashboard.tsx]: Agreed — presence of a frame proves nothing about the history: after a reconnect the live slice can be just the new `awaitingHuman` while the `gateEvaluated` was recorded before the socket came up. Fixed in 0794fb5: the any-frame shortcut is gone; the once-per-id guard alone bounds the requests (still O(open gates)), and `hydrate` merges the live frame by fingerprint so it is never double-counted. Pinned in `tests/CenterDashboard.build.test.tsx` — store holding only the live `awaitingHuman`, mocked durable log ⇒ `getRunEvents` called exactly once, the G4 verdict renders, and the live frame appears once in the merged log.

**Comment by mikeparcewski** [comment]
Re Copilot's fifth pass — the thread is fixed and resolved (0794fb5); the two suppressed remarks are taken in the same commit:

1. **Effect dependency** — the backfill effect now depends on `openGates` only and reads the store inside, so a busy dashboard does not rescan its gates on every structured frame (the once-per-id guard already made re-runs no-ops; now they do not happen).
2. **Denial field casing** — `gateVerdictModel` reads the structured denial in both spellings (`claimId`/`claim_id`, `ruleIds`/`rule_ids`, `deniedTool`/`denied_tool`), the same tolerance `denialCopy.ts` keeps. Pinned in `tests/gateVerdictModel.test.ts`.

Lint / typecheck / full suite (256 files, 2777 tests) / build green locally; PR body updated.

File: src/components/CenterDashboard.tsx [hunk 4]
```diff
@@ -963,6 +984,37 @@ export function CenterDashboard({
...
+  // by fingerprint, so nothing is double-counted. Depends on `openGates` only (the store is read
+  // inside), so a busy dashboard does not rescan its gates on every structured frame. Degrades
+  // silently — an api surface without `getRunEvents`, a 503 (no event-log binding) or an empty
+  // history leaves the card promptless, never wrong.
+  const gateBackfilled = useRef<Set<string>>(new Set());
+  useEffect(() => {
+    for (const g of openGates) {
+      const id = g.runId;
+      if (gateBackfilled.current.has(id)) continue;
+      gateBackfilled.current.add(id);
```
[the remark this answers is not in the document]
Comment (mikeparcewski) [reply src/components/CenterDashboard.tsx]: Agreed — a run can open more than one gate while the dashboard stays mounted (a retry of the same ord included), and keyed by run id alone the guard would skip the second one. Fixed in 6270a39: the guard is now the gate instance, `runId:ord:receivedAt`, so each newly opened gate refreshes its durable prefix exactly once and a re-set of the same gate fetches nothing; the budget stays O(gates opened), never O(rows). Pinned in `tests/CenterDashboard.build.test.tsx`: same gate re-set ⇒ still 1 call; second gate on the same run (later `receivedAt`) ⇒ exactly 2.

File: src/components/CenterDashboard.tsx [hunk 5]
```diff
@@ -963,6 +984,41 @@ export function CenterDashboard({
...
+    for (const g of openGates) {
+      const id = g.runId;
+      const instance = `${id}:${g.ord}:${g.receivedAt}`;
+      if (gateBackfilled.current.has(instance)) continue;
+      gateBackfilled.current.add(instance);
+      try {
+        api
+          .getRunEvents(id)
+          .then(({ events }) => useRunEventStore.getState().hydrate(id, events))
+          .catch(() => { /* no event-log binding, or no persisted history — no backfill */ });
```
[the remark this answers is not in the document]
Comment (mikeparcewski) [reply src/components/CenterDashboard.tsx]: Agreed — the per-instance key fetched fresh history, but the card rendered from the previous slice while that fetch was pending, and an empty history / 503 left it that way. Fixed in f4dac95: readiness is tracked per gate instance (`runId:ord:receivedAt`) and the inbox card renders a verdict ONLY once that instance's fetch has resolved WITH history (durable prefix merged). Pending ⇒ no block; empty history ⇒ stays un-ready, no block; 503 / no event-log binding ⇒ stays un-ready, no block — never the stale slice, which is the contract the comment states. Pinned in `tests/CenterDashboard.build.test.tsx`: a same-ord retry after a reconnect with a deferred fetch renders no block while pending and the retry's own `repo_checks` denial once the log lands (attempt 0's worktree-guard denial never appears); empty history ⇒ no block; rejected fetch ⇒ no block; a second instance on the same run proves its own history before its block renders.
